app/prepare_proposal.go

```diff
@@ -14,11 +14,9 @@ import (
 // generates the data root for the proposal block and passes it back to
 // tendermint via the BlockData.
 func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
-	sdkCtx, err := app.NewProcessProposalQueryContext()
-	if err != nil {
-		panic(err)
-	}
-
+	// create a context using a branch of the state and loaded using the
+	// proposal height and chain-id
+	sdkCtx := app.NewProposalContext(core.Header{ChainID: app.GetChainID(), Height: app.LastBlockHeight() + 1})
```

Comment:
for posteriety, if we don't do this, then the signatures of some txs will get rejected because the chain-id and height is not consistently loaded into the context, resulting in `PrepareProposal` filtering out those transactions, and therefore those txs get caught in the mempool, which always produces confusing errors.